doc/nrf/device_guides/working_with_nrf/nrf53/qspi_xip_guide.rst

```diff
@@ -0,0 +1,324 @@
+            - end
+        region: sram_primary
+        size: 0x10000
+    settings_storage:
+        address: 0xf0000
+        end_address: 0x100000
+        region: flash_primary
+        size: 0x10000
+    sram_primary:
+        address: 0x20002000
+        end_address: 0x20070000
+        region: sram_primary
+        size: 0x6e000
+
+Linker script
+=============
+
+To relocate code to the external flash, you need to configure a linker script.
+The script needs to describe the ``EXTFLASH`` flash memory block to which the code will be linked.
+The ``ORIGIN`` of the area can be calculated using following elements:
+
+* The QSPI memory starting with the 0x10000000 internal memory address.
+* The offset of an external application part image within the QSPI flash.
+  The external application code partition is mapped by the ``mcuboot_primary_2`` PM partition.
+* The image header size of MCUboot (0x200).
+
+See the following example of the calculation:
+
+.. code-block:: console
+
+    #include <zephyr/linker/sections.h>
+    #include <zephyr/devicetree.h>
+    #include <zephyr/linker/linker-defs.h>
+    #include <zephyr/linker/linker-tool.h>
+
+    MEMORY
+    {
+        /* This maps in mcuboot_primary_2 partition defined in pm_static.yaml
+        * components for ORIGIN calculation:
+        *  - 0x10000000: offset of QSPI external memory in SoC memory mapping.
+        *  - 0x120000: mcuboot_primary_2 offset in QSPI external memory
+        *  - 0x200: image header size.
+        * The size of this region is size of mcuboot_primary_2 reduced by the
+        * image header size.
+        */
+        EXTFLASH (wx) : ORIGIN = 0x10120200, LENGTH = 0x3FE00
+    }
+
+    #include <zephyr/arch/arm/cortex_m/scripts/linker.ld>
+
+Code relocation setup
+=====================
+
+Relocating code to QSPI XIP is a part of the project's :file:`CMakeLists.txt` file.
+You can set up the relocation on a file or library basis using the ``zephyr_code_relocate()`` function.
+For example, to relocate a file in the application, you can use the following configuration:
+
+.. code-block:: console
+
+   zephyr_code_relocate(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth.c LOCATION EXTFLASH_TEXT NOCOPY)
+   zephyr_code_relocate(FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/bluetooth.c LOCATION RAM_DATA)
+
+Where the first line relocates the XIP code (.text) and the second line relocates the data initialization content section (.data).
+
+Similarly, it is possible to relocate certain libraries, for example:
+
+.. code-block:: console
+
+   zephyr_code_relocate(LIBRARY subsys__mgmt__mcumgr__mgmt LOCATION EXTFLASH_TEXT NOCOPY)
+   zephyr_code_relocate(LIBRARY subsys__mgmt__mcumgr__mgmt LOCATION RAM_DATA)
+
+Flashing the project
+====================
+
+For the nRF5340 DK and other boards equipped with flash working in the QSPI mode, use the `west flash` command.
+For other cases, flashing needs to be done manually.
+
+Flashing to external flash in SPI/DSPI mode
```

Comment:
@nvlsianpu The doc comment says we should omit mentions about qspi.ini file and the weather station, which would probably also mean the same for smp_svr, as the further information depends on whether user customizes the qspi.ini file or not.
Is my understanding correct and is this paragraph all that needs to be transferred?

Reply:
When flashing applications using west, this will invoke the nrfjprog runner. This runner will use the system default configuration that will configure the application in the QSPI mode (when flashing the external flash). This behavior can be changed using a custom Qspi.ini configuration file, but this will prevent flashing from being performed using west. Such a file is needed  to work on Nordic Thingy:53. If you decide to use the Qspi.ini file, the HEX files in the repository need to be manually flashed. For example, for the smp_svr application, the files to flash are the following (paths are relative to the build directory):

hci_rpmsg/zephyr/merged_CPUNET.hex
mcuboot/zephyr/zephyr.hex
zephyr/internal_flash_signed.hex
zephyr/qspi_flash_signed.hex
The follow commands can be used to flash and verify the smp_svr sample:

nrfjprog -f NRF53 --coprocessor CP_NETWORK --sectorerase --program hci_rpmsg/zephyr/merged_CPUNET.hex --verify
nrfjprog -f NRF53 --sectorerase --program mcuboot/zephyr/zephyr.hex --verify
nrfjprog -f NRF53 --sectorerase --program zephyr/internal_flash_signed.hex --verify
nrfjprog -f NRF53 --qspisectorerase --program zephyr/qspi_flash_signed.hex --qspiini <path_to>/Qspi.ini --verify
nrfjprog -f NRF53 --reset

Note: The external flash chip must be connected to the dedicated QSPI peripheral port pins of the nRF5340, it is not possible to program an external flash chip connected to different pins using nrfjprog.

Reply:
I would generalize the commanding snippet so it will be more generic (essentialy cpunet programing part)

Reply:
added - thank you!